util/math_reference.cpp

```diff
@@ -26,11 +26,6 @@ void type_punn(const A &from, B &to) {
               reinterpret_cast<const void *>(&from), sizeof(A));
 }
 
-template <typename T>
-int32_t num_bits(T) {
-  return int32_t(sizeof(T) * 8u);
-}
-
 const uint64_t max_uint64_t = (~0x0ull);
 const uint32_t max_uint32_t = 0xffffffff;
 const uint16_t max_uint16_t = 0xffff;
```

Comment:
In the future, while modernizing the CTS we should replace these by https://en.cppreference.com/w/cpp/types/numeric_limits